cloud/deployment/workerqueue/workerqueue.go

```diff
@@ -0,0 +1,165 @@
+package workerqueue
+
+import (
+	"fmt"
+	"io"
+
+	"github.com/astronomer/astro-cli/astro-client"
+	"github.com/astronomer/astro-cli/cloud/deployment"
+	"github.com/pkg/errors"
+)
+
+var (
+	ErrWorkerQueueDefaultOptions = errors.New("failed to get worker-queue default options")
+	ErrInvalidWorkerQueueOption  = errors.New("worker-queue option is invalid")
+	ErrCannotUpdateExistingQueue = errors.New("worker-queue exists")
+)
+
+func Create(ws, deploymentID, name string, isDefaultWQueue bool, wQueueMin, wQueueMax, wQueueConcurrency int, client astro.Client, out io.Writer) error {
+	var (
+		requestedDeployment       astro.Deployment
+		err                       error
+		errHelp                   string
+		workerQueueToCreate       *astro.WorkerQueue
+		wQueueListToCreate        []astro.WorkerQueue
+		workerQueueDefaultOptions astro.WorkerQueueDefaultOptions
+		pool                      astro.NodePool
+		nodePoolIDForWorkerQueue  string
+	)
+	// get or select the deployment
+	requestedDeployment, err = deployment.GetDeployment(ws, deploymentID, client)
+	if err != nil {
+		return err
+	}
+
+	// TODO should we only get defaults once via init()
+	// get defaults for min-count, max-count and concurrency from API
+	workerQueueDefaultOptions, err = GetWorkerQueueDefaultOptions(client)
+	if err != nil {
+		return err
+	}
+
+	// Get the default nodepool ID to use
+	for _, pool = range requestedDeployment.Cluster.NodePools {
+		if pool.IsDefault {
+			nodePoolIDForWorkerQueue = pool.ID
```

Comment:
```suggestion
			nodePoolIDForWorkerQueue = pool.ID
			break
```
assuming there would be just one default

Reply:
@neel-astro that is correct and I will break out of the for loop once I find the default node pool.